data_safe_haven/commands/users.py

```diff
@@ -146,15 +146,27 @@ def register(
 
         # List users
         users = UserHandler(context, graph_api)
-        available_usernames = users.get_usernames_entra_id()
+        available_users = users.entra_users.list()
+        user_dict = {
+            user.preferred_username.split("@")[0]: user.preferred_username.split("@")[1]
+            for user in available_users
+        }
         usernames_to_register = []
         for username in usernames:
-            if username in available_usernames:
-                usernames_to_register.append(username)
+            if username in user_dict.keys():
```

Comment:
```suggestion
            if username in user_dict:
```

This should be equivalent.